packages/eslint-plugin-tslint/tsconfig.build.json

```diff
@@ -1,6 +1,9 @@
 {
   "extends": "../../tsconfig.base.json",
   "compilerOptions": {
+    // specifically disable declarations for the plugin
```

Comment:
The code clearly does that - a more interesting comment IMO would be _why_. Which is a question I have - do we not want folks programmatically accessing anything from `eslint-plugin-tslint`?

Reply:
The why specifically is commented here:
https://github.com/typescript-eslint/typescript-eslint/pull/6458/files#diff-e5831196ebbae7f2aa0010959e8dad69be30d30a0b26e691d2cf0bc332230f14R1

I'll move the comment to a more useful spot.

> We purposely don't generate types for our plugin because:
> 1) there's no real reason that anyone should do a typed import of our rules,
> 2) it would require us to change our code so there aren't as many inferred types

You can try it yourself and see the errors - for TS to generate declarations we need to annotate all of our `export default createRule(...)` like `export default createRule(...) satisfies RuleModule<TMessageIds, TOptions>`. In a nutshell - TS knows that to generate the `.d.ts` file it'll need that `RuleModule` type - but TS doesn't like to insert an import on your behalf, so it errors asking you to annotate it so that the `.d.ts` generation remains just an elision process.